### PR TITLE
feat(api): improve nodejs api

### PR DESCRIPTION
**Pull request by eduardomourar**
In order to make somewhat more extensible to use library with Node.js, we will need to make the following changes:

* Allow the input to be an object (a single schema or multiple schemas either with parsed content or its full path)
* Make the output an plain object containing the readme, input schemas and markdown
* Create TypeScript definition

Related to #300.

### CODE REVIEW

File: lib/formatInfo.js [hunk 1]
```diff
@@ -49,24 +49,24 @@ function iscustom(schema) {
 function getdefined(schema) {
   if (schema[s.parent]) {
     return {
-      text: `${path.basename(schema[s.filename])}*`,
```
Comment (trieloff) [inline lib/formatInfo.js:52]: What's the background of this change? It seems unrelated.

File: lib/formattingTools.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
-/*
- * Copyright 2019 Adobe. All rights reserved.
```
Comment (trieloff) [inline lib/formattingTools.js:2]: Keep the date

File: lib/keywords.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
-/*
- * Copyright 2019 Adobe. All rights reserved.
```
Comment (trieloff) [inline lib/keywords.js:2]: Date

**Review by trieloff — CHANGES_REQUESTED** [review]
Thank you for the PR, but I find it too hard to review the unrelated changes and disentangle the substance of the PR. 

Please:

- do not change the copyright dates
- do not introduce camel case variable names (open a separate refactoring/style PR if it bothers you too much)
- do not change import orders or import names (same as above)
- make sure the test coverage does not go down

I will review then, as the substance of the PR seems to make sense, it's just hard to see underneath the mountain of other changes.

File: .eslintrc.js [hunk 4]
```diff
@@ -49,8 +49,8 @@ module.exports = {
     }],
 
     // enforce license header (todo: improve plugin to support patterns for multi-lines)
-    'header/header': [2, 'block', ['',
-      ' * Copyright 2019 Adobe. All rights reserved.',
+    'header/header': [2, 'block', ['*',
+      ' * Copyright 2021 Adobe. All rights reserved.',
```
[the remark this answers is not in the document]
Comment (eduardomourar) [reply .eslintrc.js:53]: i ran into issues with eslint and having two sets of headers, because it will use this template to fix all files. is there are a way to solve it?

File: README.md [hunk 5]
```diff
@@ -89,7 +89,7 @@ If you run `npm install` before running `npm run prepare`, `npm` will install th
 ### With API for Node.js
 
 ```javascript
-const jsonschema2md = require('@adobe/jsonschema2md');
+const { jsonschema2md } = require('@adobe/jsonschema2md');
```
[the remark this answers is not in the document]
Comment (eduardomourar) [reply README.md:92]: it is just doc update. in reality this section needs some improvement and more examples

File: .eslintrc.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (trieloff) [reply .eslintrc.js:53]: Yes, let me take a look, I think we can specify a range and use the current year as the default.

[the remark this answers is not in the document]
Comment (trieloff) [reply .eslintrc.js:53]: Should be fixed in `main` now.

File: lib/markdownBuilder.js [hunk 6]
```diff
@@ -853,7 +853,7 @@ function build({
 
   console.log('generating markdown');
   return (schemas) => foldl(schemas, {}, (pv, schema) => {
-    const slugger = ghslugger();
+    const slugger = new GhSlugger();
```
Comment (eduardomourar) [inline lib/markdownBuilder.js:856]: using new keyword is needed based on `@types/github-slugger` and, in consequence, the case change is required.

File: lib/formatInfo.js (hunk 1 above)
Comment (eduardomourar) [reply lib/formatInfo.js:52]: `filename` symbol now holds the basename and `fullpath` symbol holds the full path of the file (not always available)

**Comment by trieloff** [comment]
Hi @eduardomourar - thank for the update, I may need until the week after next to review this.

**Comment by eduardomourar** [comment]
> Hi @eduardomourar - thanks for the update, I may need until the week after next to review this.

No rush, we are doing extensive tests against our schemas in the meantime.

File: .eslintrc.js [hunk 7]
```diff
@@ -49,7 +49,7 @@ module.exports = {
     }],
 
     // enforce license header (todo: improve plugin to support patterns for multi-lines)
-    'header/header': [2, 'block', ['',
+    'header/header': [2, 'block', ['*',
```
Comment (trieloff) [inline .eslintrc.js:52]: This is not correct, it turns the license comment from a normal block comment `/*` into a JSDoc comment `/**`

File: test/formatting.test.js [hunk 8]
```diff
@@ -1,5 +1,5 @@
-/*
- * Copyright 2019 Adobe. All rights reserved.
+/**
+ * Copyright 2021 Adobe. All rights reserved.
```
Comment (trieloff) [inline test/formatting.test.js:2]: These dates should not be updated.

**Review by trieloff — CHANGES_REQUESTED** [review]
Good, but the comment formatting and copyright dates need to stay the way they've been before.

File: .eslintrc.js (hunk 7 above)
Comment (eduardomourar) [reply .eslintrc.js]: i put this because typescript compiler only considers JSDoc, so it will remove most of the license headers

**Review by trieloff — APPROVED** [review]
